src/server/game/Spells/Auras/SpellAuraEffects.cpp

```diff
@@ -4037,28 +4037,23 @@ void AuraEffect::HandleAuraModIncreaseEnergyPercent(AuraApplication const* aurAp
         return;
 
     Unit* target = aurApp->GetTarget();
-
     Powers powerType = Powers(GetMiscValue());
-    // do not check power type, we can always modify the maximum
-    // as the client will not see any difference
-    // also, placing conditions that may change during the aura duration
-    // inside effect handlers is not a good idea
-    //if (int32(powerType) != GetMiscValue())
-    //    return;
-
     UnitMods unitMod = UnitMods(UNIT_MOD_POWER_START + powerType);
-    float amount = float(GetAmount());
 
-    if (apply)
-    {
-        target->HandleStatModifier(unitMod, TOTAL_PCT, amount, apply);
-        target->ModifyPowerPct(powerType, amount, apply);
-    }
-    else
-    {
-        target->ModifyPowerPct(powerType, amount, apply);
-        target->HandleStatModifier(unitMod, TOTAL_PCT, amount, apply);
-    }
+    float pct = float(GetAmount());
+
+    // Calculate the change.
+    int32 change = target->GetMaxPower(powerType);
+    ApplyPct(change, apply ? pct : (100 - 100 / (1 + pct / 100))); // On removal (!apply) GetMaxPower is already increased by pct.
```

Comment:
what does this calculation change?

Reply:
increases pct for pct 😃 because base (max power, line above) is already increased by pct

Reply:
can you explain that is clearlier

Reply:
Imagine u have +30% max mana, lets say its 13000. Old code would reduce power by 30% of current max (before effect is removed), thats 13000*0,30 = 39k, correct should be 30k. I hope its clear enough :P

Reply:
Sorry, 3,9k, correct should be 3k*

Reply:
in your example i get a change of -28.8k

Reply:
https://www.onlinegdb.com/r1kMH0GzM

Reply:
Another valid formula is `pct * 100 / (pct + 100)` - same as `100 - 100 / (1 + pct / 100)` ... But then it depends on how many times we want to use pct in a formula.

Reply:
i still don't get what was wrong with `ModifyPowerPct`, if there is something wrong, `ApplyPercentModFloatVar` must be wrong too

Reply:
there's nothing wrong with `ModifyPowerPct`, it's wrong that it reduces max by pct: reducing 13000 for 30% should get us back to 10000, but 13000 * 0.30 =/= 3000 - because we have to reduce for 3000, the same amount of power we gave as we increased from 10000 to 13000 - 3000! 😄 instead that 3000 is alot bigger, 3900.

Another example from 3.3.5:
At 30k mana, priest casting hymn of hope (I believe it's 30% for the priest, 15% for members, but it doesn't matter) would get **9k max mana + 9k mana**, but at the end of effect he would lose **11.7k mana + 9k max mana**. See what's wrong?